src/main/java/com/nvidia/spark/rapids/jni/DecimalUtils.java

```diff
@@ -25,11 +25,32 @@ public class DecimalUtils {
     NativeDepsLoader.loadNativeDeps();
   }
 
+  /**
+   * Multiply two DECIMAL128 columns together into a DECIMAL128 product rounded to the specified
+   * scale with overflow detection. This method considers a precision greater than 38 as overflow
+   * even if the number still fits in a 128-bit representation.
+   * @param a factor input, must match row count of the other factor input
+   * @param b factor input, must match row count of the other factor input
+   * @param productScale scale to use for the product type
+   * @return table containing a boolean column and a DECIMAL128 product column of the specified
+   *         scale. The boolean value will be true if an overflow was detected for that row's
+   *         DECIMAL128 product value. A null input row will result in a corresponding null output
+   *         row.
+   */
+  public static Table mul128(ColumnView a, ColumnView b, int productScale) {
```

Comment:
Can we please just expose the boolean instead of having it hidden with `mul128` vs `multiply128`. It is too subtle of a name change to stand out if there is a bug. Or name them `multiply128Corrected` and then `multiply128SparkLegacy`, but keep the original `multiply128` until the plugin is updated.

Reply:
Originally I wanted to pass in a boolean but talked myself out of it because I thought passing a boolean was not very pretty and ruled in favor of having a separate method. But coming up with a name was challenging and considering that you are raising this, I think I will just pass a boolean parameter.